tests/functional/api/test_objectstorage.py

```diff
@@ -275,6 +275,72 @@ def test_container_set_properties(self):
             exc.NoSuchContainer, self.api.container_set_properties,
             self.account, name, metadata)
 
+    def test_container_flush(self):
+        name = random_str(32)
+
+        # no container
+        self.assertRaises(exc.NoSuchContainer,
+                          self.api.container_flush, self.account, name)
+
+        def flush_and_check():
+            self.api.container_flush(self.account, name)
+            properties = self.api.container_get_properties(self.account, name)
+            self.assertEqual(properties['system']['sys.m2.objects'], '0')
+            self.assertEqual(properties['system']['sys.m2.usage'], '0')
+            list_objects = self.api.object_list(self.account, name)
+            self.assertEqual(len(list_objects['objects']), 0)
+
+        # empty container
+        self.api.container_create(self.account, name)
+        flush_and_check()
+
+        # one content
+        self.api.object_create(self.account, name, obj_name='content',
+                               data='data')
+        flush_and_check()
+
+        # many contents
+        for i in range(1024):
+            self.api.object_create(self.account, name,
+                                   obj_name='content'+str(i), data='data')
+        flush_and_check()
+
+        # give meta2 and beanstalk time to catch his breath
+        time.sleep(10)
+
+    def test_container_flush_fast(self):
+        name = random_str(32)
+
+        # no container
+        self.assertRaises(exc.NoSuchContainer,
+                          self.api.container_flush, self.account, name, True)
+
+        def flush_and_check():
+            self.api.container_flush(self.account, name, fast=True)
+            properties = self.api.container_get_properties(self.account, name)
+            self.assertEqual(properties['system']['sys.m2.objects'], '0')
+            self.assertEqual(properties['system']['sys.m2.usage'], '0')
+            list_objects = self.api.object_list(self.account, name)
+            self.assertEqual(len(list_objects['objects']), 0)
+
+        # empty container
+        self.api.container_create(self.account, name)
+        flush_and_check()
+
+        # one content
+        self.api.object_create(self.account, name, obj_name='content',
+                               data='data')
+        flush_and_check()
+
+        # many contents
+        for i in range(1024):
+            self.api.object_create(self.account, name,
+                                   obj_name='content'+str(i), data='data')
+        flush_and_check()
+
+        # give meta2 and beanstalk time to catch their breath
+        time.sleep(30)
```

Comment:
Couldn't we wait for the scores to raise a little instead of arbitrary sleeping this huge amount of time?